### PR TITLE
fix(developer): refactor kmcmplib compiler messages to use map

### PR DESCRIPTION
**Pull request by markcsinclair**
Refactor CompMsg.cpp to use a map, which removes the need for linear searching in `GetCompilerErrorString`

Supporting tests added in `gtest-compmsg-test.cpp`

Removed from PR [#11378](https://github.com/keymanapp/keyman/pull/11378)

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by markcsinclair** [comment]
Not sure why this closed when I merged PR [#11378](https://github.com/keymanapp/keyman/pull/11378) into master

**Comment by markcsinclair** [comment]
Merge conflicts resolved, ready to review

**Comment by mcdurdin** [comment]
> Not sure why this closed when I merged PR [#11378](https://github.com/keymanapp/keyman/pull/11378) into master

The commit message was `moved CompMsg.cpp refactor to fix #11738`. GitHub saw `fix #11738`...

**Review by ermshiperete — APPROVED** [review]

File: developer/src/kmcmplib/src/CompMsg.cpp [hunk 1]
```diff
@@ -150,14 +146,8 @@ const struct CompilerError CompilerErrors[] = {
...
 
-KMX_CHAR *GetCompilerErrorString(KMX_DWORD code)
-{
-  for(int i = 0; CompilerErrors[i].ErrorCode; i++) {
-    if(CompilerErrors[i].ErrorCode == code) {
-      return ( KMX_CHAR*) CompilerErrors[i].Text;
-    }
-  }
-  return nullptr;
+KMX_CHAR *GetCompilerErrorString(KMX_DWORD code) {
```
Comment (mcdurdin) [inline developer/src/kmcmplib/src/CompMsg.cpp:151]: next step, make this return a `const KMX_CHAR *` ... so much `const` missing in kmcmplib.

**Review by mcdurdin — APPROVED** [review]

**Review by ermshiperete — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.57-alpha](https://keyman.com/downloads/releases/alpha/18.0.57)

**Pull request merged**